### ISSUE TITLE
[AutoDeploy] Arch2: Model Support: VLM, Long-Context, and Linear Attention

### ISSUE THREAD

**Issue by sugunav14** [NVIDIA/TensorRT-LLM#4593]
A running epic to track support for increasing model coverage beyond attention-based text-to-text models

Some imminent features / discussions for that entail:

* Add support for handling logit soft capping that is used in (used in Gemini, Grok and Gemma-2, etc.) The attention system currently fails for google/gemma-2-27b-it. ("warm-up")
* Considering linear attention and other state-space model approaches
* Long-context and seqlen-dependent attention masking (sliding window, chunked attention, ...)
* VLM and other many-to-text models

**Comment by lucaslie** [comment]
For this epic, we should start out by re-surveying the list of models from popular sources like Huggingface to understand what are the most impactful/downloaded/used models beyond our current scope. 

Current scope: text-only, auto-regressive, quadratic attention.

From the list of models/model classes we can then derive a set of features needed inside AutoDeploy to support these models and we can then prioritize accordingly.

Examples of potential new models/model classes:
* Llama4 or Qwen: text+image to image
* Llama4, Gemma, .... : sliding window, chunked attention window, and other long-context attention windows
* Nemotron-H: Linear attention, state-space models, mamba, etc.... 
* ... ? 

Examples of corresponding features:
* logit softcap to support a special type of attention as used in Gemma 2
* sliding window support 
* ....
